sns/parity_math.c

```diff
@@ -311,12 +609,47 @@ M0_INTERNAL int m0_parity_math_init(struct m0_parity_math *math,
 
 	M0_SET0(math);
 
+	M0_ENTRY("data_count=%u parity_count=%u", data_count, parity_count);
+
 	math->pmi_data_count	= data_count;
 	math->pmi_parity_count	= parity_count;
 
 	if (parity_count == 1) {
 		math->pmi_parity_algo = M0_PARITY_CAL_ALGO_XOR;
-		return 0;
+		return M0_RC(ret);
```

Comment:
if (parity_count == 1) {
 ..
 return ret;
 ..
}
#if ISAL_ENCODE_ENABLED

#else

#end

i.e else statements can be avoided.

Reply:
Done